src/infra/redis/ioredis.adapter.ts

```diff
@@ -0,0 +1,197 @@
+import { Redis } from 'ioredis';
+import { Logger } from '../logger/index.js';
+import { addMessageCommand, xDelIfEmptyCommand } from './commands.js';
+import { Task, XAutoClaimResponse } from './interfaces/redis.interface.js';
+import { StreamMessageReply, StreamsMessagesReply } from './interfaces/stream-message-reply.js';
+import { StreamNameClockPair } from './interfaces/stream-name-clock-pair.js';
+import { mapToStreamMessagesReplies, mapToStreamsMessagesReply, mapToXAutoClaimResponse } from './mapper.js';
+import { RedisAdapter } from './redis.adapter.js';
+import { RedisConfig } from './redis.config.js';
+
+export class IoRedisAdapter implements RedisAdapter {
+	public readonly redisPrefix: string;
+	public readonly redisDeleteStreamName: string;
+	public readonly redisWorkerStreamName: string;
+	public readonly redisWorkerGroupName: string;
+	public readonly redisDeletionActionKey: string;
+
+	public constructor(
+		private readonly internalRedisInstance: Redis,
+		private readonly config: RedisConfig,
+		private readonly logger: Logger,
+	) {
+		this.logger.setContext(IoRedisAdapter.name);
+
+		this.redisPrefix = this.config.REDIS_PREFIX;
+		this.redisDeleteStreamName = `${this.redisPrefix}:delete`;
+		this.redisWorkerStreamName = `${this.redisPrefix}:worker`;
+		this.redisWorkerGroupName = `${this.redisPrefix}:worker`;
+		this.redisDeletionActionKey = `${this.redisPrefix}:delete:action`;
+
+		internalRedisInstance.defineCommand('addMessage', {
+			numberOfKeys: 1,
+			lua: addMessageCommand(this.redisWorkerStreamName, this.redisWorkerGroupName),
+		});
+
+		internalRedisInstance.defineCommand('xDelIfEmpty', {
+			numberOfKeys: 1,
+			lua: xDelIfEmptyCommand(),
+		});
+	}
+
+	public subscribeToDeleteChannel(callback: (message: string) => void): void {
+		this.internalRedisInstance.subscribe(this.redisDeletionActionKey);
+		this.internalRedisInstance.on('message', (chan, message) => {
+			callback(message);
+		});
+	}
+
+	public async markToDeleteByDocName(docName: string): Promise<void> {
+		await this.internalRedisInstance.xadd(this.redisDeleteStreamName, '*', 'docName', docName);
+		await this.internalRedisInstance.publish(this.redisDeletionActionKey, docName);
+	}
+
+	public async addMessage(key: string, message: unknown): Promise<void> {
+		// @ts-ignore
```

Comment:
Was genau wird ignoriert? Können wir hier spezifische Ausnahmen verwenden?

Reply:
Wenn man die Zeile entfernt, kommt folgende Fehlermeldung in der darauffolgenden Zeile :
```
Property 'addMessage' does not exist on type 'Redis'.ts(2339)
```

Reply:
Ok Problem kommt mit  redis.defineCommand('addMessage', und redis.defineCommand('xDelIfEmpty'.
Das Interface import { Redis } from 'ioredis'; stimmt nicht mehr. Die Erweiterung müsste hinter 
const redisInstance = new Redis(redisUrl); erfolgen und die Methode muss dann ein Interface wiedergeben welches
dann interface XXX extends Redis {
addMessage
xDelIfEmpty
}
hat. Todo für refactoring nach Release.